src/lib/controllers/ui-controller.js

```diff
@@ -801,6 +814,7 @@ export default class UIController {
       this.panel.panelData.grammarRes = urls[0]
     } else {
       this.panel.panelData.grammarRes = { provider: this.l10n.messages.TEXT_NOTICE_GRAMMAR_NOTFOUND }
+      this.panel.initGrammarTab()
```

Comment:
`updateGrammar()` is called by a resource query only after grammar data is retrieved. Do we need to call `initGrammarTab()` here which purpose is to probably retrieve it again? Isn't grammar retrieval already handled by a resource query that initiated an `updateGrammar()` call?